### ISSUE TITLE
emailjs: module issues with "no form of authorization supported"

### ISSUE THREAD

**Issue by WilliamDrive** [sirensolutions/sentinl#140]
Hi, 

   I installed the sentinl 5.2.2-1 version since my kibana is 5.2.2 and created new watcher. the new alarms were created without problem. 
   but i could not see the email alert. and I tried to find the configuration for email settings in kibana.yml file.  it seems that there is no email settings. 
   where could I find those email settings?

thanks,
William

**Comment by SergeyBondarenko** [comment]
Hi @WilliamDrive 
You have to put email settings inti kibana.yml, look our [WiKi](https://github.com/sirensolutions/sentinl/wiki/SENTINL-Config-Example), for example:
```
sentinl:
  settings:
    email:
      active: true
      user: anUser@gmail.com
      password: aPassword
      host: smtp.gmail.com
      ssl: false
```
If you use Gmail, you need to [enable unsafe apps](https://support.google.com/accounts/answer/6010255?hl=en).
If you have another mail server, for example, a locally configured postfix, test if it works, Linux way:
```
echo "hello trex" | mail -s "mail for trex" receiver@gmail.com -- -f sender@localhost.it
```
Please attach the watcher config, you can get it from the Raw tab.

**Comment by WilliamDrive** [comment]
Hi @SergeyBondarenko 

     I added the email settings into kibana.yml. and restart kibana.  and there is an error found in kibana.log:
     
    "{"type":"log","@timestamp":"2017-06-09T16:34:00Z","tags":["status","info","Senti
nl","email"],"pid":34621,"level":"error","message":"connection encountered an er
ror","error":{"message":"connection encountered an error","name":"Error","stack"
:"Error: connection encountered an error\n    at module.exports (/opt/bitnami/ki
bana/plugins/sentinl/node_modules/emailjs/smtp/error.js:3:13)\n    at TLSSocket.
error (/opt/bitnami/kibana/plugins/sentinl/node_modules/emailjs/smtp/response.js
:26:29)\n    at emitOne (events.js:101:20)\n    at TLSSocket.emit (events.js:188
:7)\n    at connectErrorNT (net.js:1022:8)\n    at _combinedTickCallback (intern
al/process/next_tick.js:74:11)\n    at process._tickDomainCallback (internal/pro
cess/next_tick.js:122:9)","code":5}}"

it seems the network issue and I am using Bitnami ELK stack on Microsoft Azure, could that be a reason?

btw, I attached watcher config  for the reference. 
[watcherconfig.txt](https://github.com/sirensolutions/sentinl/files/1064899/watcherconfig.txt)

**Comment by fbaligand** [comment]
As error speaks about TLS, maybe you could try with ssl=false or ssl=true. 

Then, your mail server needs to be accessible without a proxy. 

Finally, have you tried to send a mail to your smtp configuration using another mean (like Linux command) to check that configuration is right ?

**Comment by WilliamDrive** [comment]
Hi fbaligand,

     I updated the my email settings and now I got the new error"no form of authorization supported " as following:
    {"type":"log","@timestamp":"2017-06-11T20:13:00Z","tags":["status","info","Senti
nl","email"],"pid":20827,"level":"error","message":"no form of authorization sup
ported","error":{"message":"no form of authorization supported","name":"Error","
stack":"Error: no form of authorization supported\n    at module.exports (/opt/b
itnami/kibana/plugins/sentinl/node_modules/emailjs/smtp/error.js:3:13)\n    at i
nitiate (/opt/bitnami/kibana/plugins/sentinl/node_modules/emailjs/smtp/smtp.js:5
36:44)\n    at caller (/opt/bitnami/kibana/plugins/sentinl/node_modules/emailjs/
smtp/smtp.js:48:14)\n    at attempt (/opt/bitnami/kibana/plugins/sentinl/node_mo
dules/emailjs/smtp/smtp.js:415:14)\n    at caller (/opt/bitnami/kibana/plugins/s
entinl/node_modules/emailjs/smtp/smtp.js:48:14)\n    at response (/opt/bitnami/k
ibana/plugins/sentinl/node_modules/emailjs/smtp/smtp.js:345:13)\n    at caller (
/opt/bitnami/kibana/plugins/sentinl/node_modules/emailjs/smtp/smtp.js:48:14)\n
  at response (/opt/bitnami/kibana/plugins/sentinl/node_modules/emailjs/smtp/smt
p.js:201:11)\n    at caller (/opt/bitnami/kibana/plugins/sentinl/node_modules/em
ailjs/smtp/smtp.js:48:14)\n    at Socket.response (/opt/bitnami/kibana/plugins/s
entinl/node_modules/emailjs/smtp/smtp.js:181:11)\n    at Socket.g (events.js:291
:16)\n    at emitTwo (events.js:106:13)\n    at Socket.emit (events.js:191:7)\n
   at notify (/opt/bitnami/kibana/plugins/sentinl/node_modules/emailjs/smtp/resp
onse.js:19:14)\n    at Socket.watch (/opt/bitnami/kibana/plugins/sentinl/node_mo
dules/emailjs/smtp/response.js:44:7)\n    at emitOne (events.js:96:13)\n    at S
ocket.emit (events.js:188:7)\n    at readableAddChunk (_stream_readable.js:176:1
8)\n    at Socket.Readable.push (_stream_readable.js:134:10)\n    at TCP.onread
(net.js:548:20)","code":7}}

I googled this error and it seems that emailjs fixed this issue at 3.1.0.  and the version in sentinl is 1.0.8 

should i just update package.json to upgrade the version of mailjs? 

thanks,
William

**Comment by lmangani** [comment]
@WilliamDrive sure just give it a shot by upgrading the package.json and let us know the results

**Comment by WilliamDrive** [comment]
it seems that the latest release is 1.0.9 and the current 1.0.8 version should has the fix for "no form of authorization supported " error. 
but other users also reported this issue recently.  not sure if anyone at emailjs side could fix this issue.

thanks,
William

**Comment by psameer1831** [comment]
You can fix this issue yourself by changing smtp.js in emailjs from  " **this.loggedin = (options.user && options.password) ? false: true;** " to   **this.loggedin = (options.user && options.password) ? true : false;**

emailjs is not that stable. Can we have this plugin update the email dependency to nodemailer instead ?

**Comment by lmangani** [comment]
@psameer1831 absolutely if you have a better experience with it happy to consider it

**Comment by lmangani** [comment]
Closing. Please feel free to open a separate issue to propose ```nodemailer``` integration. Thanks!

**Comment by amdawkhar** [comment]
is this fixed in 6.6.1 version, faced same issue in 6.6.1 ,changed  
**this.loggedin = (options.user && options.password) ? false: true; " to this.loggedin = (options.user && options.password) ? true : false; in smtp.js**  which is working fine
